components/server/src/api/hello-service-api.ts

```diff
@@ -14,27 +13,28 @@ import {
     SayHelloResponse,
 } from "@gitpod/public-api/lib/gitpod/experimental/v1/dummy_pb";
 import { injectable } from "inversify";
+import { ctx } from "../util/request-context";
 
 @injectable()
 export class HelloServiceAPI implements ServiceImpl<typeof HelloService> {
-    async sayHello(req: SayHelloRequest, context: HandlerContext): Promise<SayHelloResponse> {
+    async sayHello(req: SayHelloRequest, _: HandlerContext): Promise<SayHelloResponse> {
         const response = new SayHelloResponse();
-        response.reply = "Hello " + this.getSubject(context);
+        response.reply = "Hello " + getSubject();
         return response;
     }
-    async *lotsOfReplies(req: LotsOfRepliesRequest, context: HandlerContext): AsyncGenerator<LotsOfRepliesResponse> {
+    async *lotsOfReplies(req: LotsOfRepliesRequest, _: HandlerContext): AsyncGenerator<LotsOfRepliesResponse> {
         let count = req.previousCount || 0;
-        while (!context.signal.aborted) {
+        while (!ctx().signal.aborted) {
```

Comment:
I would prefer a dedicated function per aspect instead of a generic ctx() that returns everything in one shape.